backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -521,7 +526,11 @@ export class ExperimentService {
           (queries &&
             queries.length > 0 &&
             queries.map((query: any) => {
-              promiseArray.push(this.metricRepository.findOne(query.metric.key));
+              if (query.metric == null) {
+                promiseArray.push(this.metricRepository.findOne(query.query.metric.key));
```

Comment:
This looks incorrect. What is this for?

Reply:
In the `queries`, we get already created query and newly added query to be updated in this `updateExperimentInDB` function. The already created query has different doc format then the newly added one's as shown below:

`[
  {
    createdAt: '2021-11-28T09:54:41.740Z',
    updatedAt: '2021-11-28T09:54:41.740Z',
    versionNumber: 1,
    id: 'fedaff16-9654-408b-9ccc-bd55a92ed58d',
    name: 'total time',
    query: { metric: [Object], operationType: 'sum' },
    repeatedMeasure: 'MOST_RECENT',
    metricKey: null,
    experimentId: 'be3ae74f-370a-4015-93f3-7761d16f8b17'
  },
  {
    createdAt: '2021-11-28T09:54:41.740Z',
    updatedAt: '2021-11-28T09:54:41.740Z',
    versionNumber: 1,
    id: '9ba5b350-d257-4655-91f1-7ecaa360c6ab',
    name: 'time',
    query: { metric: [Object], operationType: 'avg' },
    repeatedMeasure: 'MEAN',
    metricKey: null,
    experimentId: 'be3ae74f-370a-4015-93f3-7761d16f8b17'
  },
  {
    name: 'avg time',
    query: { operationType: 'avg' },
    metric: { key: 'totalTimeSeconds' },
    repeatedMeasure: 'MOST_RECENT'
  }
]`

But I have changed this now as per what you want.